Added the carbon filter and an apply button

XiaoyiSun opened the pull request:
### Change type:
- New feature

###Description:
- Added the carbon filter (placeholder)
- Added an Apply button under Advanced Search
- Refactor filter components

XiaoyiSun committed:
Added carbon filter and apply button

XiaoyiSun committed:
removed unnecessary imports

eldss reviewed (commented):
Having a weird issue when I click on See Stories from a popup in the map.

In Firefox, I get this error:
![image](https://user-images.githubusercontent.com/29155905/76787826-fefcbe80-6776-11ea-9f0b-c6f5bed018fa.png)

In Chromium I don't get an error on the page but no stories show up and this appears in the console:
```
Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
    in input (at StorySearchBar.js:21)
    in form (at StorySearchBar.js:20)
    in div (at StorySearchBar.js:18)
    in StorySearchBar (at Stories.js:87)
    in section (at Stories.js:86)
    in Stories (at App.js:30)
    in Route (at App.js:30)
    in Switch (at App.js:19)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App.js:18)
    in App (at src/index.js:6)
```
For me this was happening with every place I clicked on. I'm not sure if this due to React code, or an issue with the places server however. Can you see if this happening for you as well, and if you can tell whether it is a bug on our end or a server error?

XiaoyiSun committed:
fixed stories search bar default value

XiaoyiSun commented:
> Having a weird issue when I click on See Stories from a popup in the map.

> For me this was happening with every place I clicked on. I'm not sure if this due to React code, or an issue with the places server however. Can you see if this happening for you as well, and if you can tell whether it is a bug on our end or a server error?

@eldss Thanks! Quick fix for these problems.

Actually, there are two problems. One on the map page, the other on the stories page.
1. in firefox, this is because the onBlur function will be run **after** we switch to the stories page. So the suggestion component is gone when querySelector runs. Fixed. However, I am thinking about the behavior of clicking on a popup on the map. Should we open a new window (instead of losing everything on the map and redirect to the stories page)?
2. I got the same warning (Warning: `value` prop on `input` should not be null.) using both chrome and firefox. This warning occurs on the stories page when no storyTitle in URL params. Fixed by setting the default value to an empty string.

Take a look! Thanks!

eldss commented:
Looks great!

I'm still not seeing any stories, but that does appear to be a problem with the database. I'm not getting anything from querying the server directly in the browser either.
